packages/models/src/agreement/agreement.ts

```diff
@@ -25,6 +25,21 @@ export const AgreementState = z.enum([
 ]);
 export type AgreementState = z.infer<typeof AgreementState>;
 
+export const agreementStampKind = {
+  submission: "submission",
+  activation: "activation",
+  rejection: "rejection",
+  suspensionByProducer: "suspensionByProducer",
+  suspensionByConsumer: "suspensionByConsumer",
+  upgrade: "upgrade",
+  archiving: "archiving",
+} as const;
+export const AgreementStampKind = z.enum([
+  Object.values(agreementStampKind)[0],
+  ...Object.values(agreementStampKind).slice(1),
+]);
+export type AgreementStampKind = z.infer<typeof AgreementStampKind>;
```

Comment:
This could also be derived from the existing `AgreementStamps` type:

```suggestion
export const AgreementStampKind = AgreementStamps.keyof();
```

Reply:
Updated, thanks!